src/qiskit_qec/utils/stim_tools.py

```diff
@@ -0,0 +1,347 @@
+            for qubit in qargs:
+                if qubit._register.name not in register_offset:
+                    register_offset[qubit._register.name] = previous_offset
+                    previous_offset += qubit._register.size
+
+            qubit_indices = [
+                qargs[i]._index + register_offset[qargs[i]._register.name]
+                for i in range(len(qargs))
+            ]
+
+            if isinstance(inst, QuantumChannelInstruction):
+                # Errors: it would be good to add exeptions for delpoarizinng noise, Z error etc,
+                # because the stim detctor error model relies on the disjoint error assumption
+                # for general Pauli channel
+                qerror = inst._quantum_error
+                pauli_errors_types = qerror.to_dict()["instructions"]
+                pauli_probs = qerror.to_dict()["probabilities"]
+                if pauli_errors_types[0][0]["name"] in pauli_error_1_stim_order:
+                    probs = 4 * [0.0]
+                    for pind, ptype in enumerate(pauli_errors_types):
+                        probs[pauli_error_1_stim_order[ptype[0]["name"]]] = pauli_probs[pind]
+                    stim_circuit.append("PAULI_CHANNEL_1", qubit_indices, probs[1:])
+                elif pauli_errors_types[0][0]["params"][0] in pauli_error_2_stim_order:
+                    # here the name is always 'pauli' and the params gives the Pauli type
+                    probs = 16 * [0.0]
+                    for pind, ptype in enumerate(pauli_errors_types):
+                        probs[pauli_error_2_stim_order[ptype[0]["params"][0]]] = pauli_probs[pind]
+                    stim_circuit.append("PAULI_CHANNEL_2", qubit_indices, probs[1:])
+                else:
+                    raise Exception("Unexpected operations: " + str([inst, qargs, cargs]))
+            else:
+                # Gates and measurements
+                if inst.name in qiskit_to_stim_dict:
+                    if len(cargs) > 0:  # keeping track of measurement indices in stim
+                        measurement_data.append(
+                            [
+                                cargs[0]._index + register_offset[qargs[0]._register.name],
+                                qargs[0]._register.name,
+                            ]
+                        )
+                    if qiskit_to_stim_dict[inst.name] == "TICK":  # barrier
+                        stim_circuit.append("TICK")
+                    else:  # gates/measurements acting on qubits
+                        stim_circuit.append(qiskit_to_stim_dict[inst.name], qubit_indices)
+                else:
+                    raise Exception("Unexpected operations: " + str([inst, qargs, cargs]))
+
+        stim_circuits[circ_label] = stim_circuit
+        stim_measurement_data[circ_label] = measurement_data
+    return stim_circuits, stim_measurement_data
+
+
+def get_counts_via_stim(circuits, shots: int = 4000, noise_model=None):
```

Comment:
Can you specific what the noise model is with typing/docstrings (same as other noise_model comment)

Reply:
done